workerpool/pool.go

```diff
@@ -58,20 +63,41 @@ func (pool *WorkerPool) Start() error {
 				for _, worker := range job.workers {
 					worker.done <- true
 				}
+
+				if err := job.mutex.Unlock(context.TODO()); err != nil {
+					pool.log.Errorf("unable to unlock", "error", err, "connectionID", job.connection.ID)
+				}
+				pool.log.Debugf("unlocked", "connectionID", job.connection.ID)
 			}
 			return nil
 		case event := <-pool.events:
-			count := event.Connection.Source.NumberOfWorkers()
-			if _, ok := pool.jobs[event.Connection.ID]; !ok {
-				pool.jobs[event.Connection.ID] = &job{
-					connection: event.Connection,
-					workers:    make(map[uint]*worker),
+			if event.Type == watcher.Created {
```

Comment:
Would we add a case here that would be `watcher.Done` or `watcher.Closed`? That should for a job to unsubscribe altogether.